src/Tribe/Views/V2/View.php

```diff
@@ -2194,50 +2194,77 @@ protected function should_show_recent_past_events_view() {
+		// Show Latest Past Events only on the default view.
 		if ( $this->get_slug() !== $default_slug ) {
 			return;
 		}
 
+		$now    = $this->context->get( 'now' );
+		$latest = tribe_events_latest_date();
+
+		// If now is less then the latest event published, do not show.
+		if ( $now < $latest ) {
+			return;
+		}
+
+
+		// Checks to verify on the initial load of a view or if using today's date for the view.
+		$today     = $this->context->get( 'today' );
+		$view_date = $this->context->get( 'event_date', '' );
+		if ( 'month' === $this->get_slug() ) {
+			$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBYEARMONTHTIMEFORMAT );
+			$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBYEARMONTHTIMEFORMAT );
+		} elseif ( 'week' === $this->get_slug() ) {
+			list( $today_week_start, $today_week_end ) = Dates::get_week_start_end( $today, (int) $this->context->get( 'start_of_week', 0 ) );
+			list( $view_week_start, $view_week_end ) = Dates::get_week_start_end( $view_date, (int) $this->context->get( 'start_of_week', 0 ) );
+			$today_formatted     = $today_week_start->format( Dates::DBDATEFORMAT );
+			$view_date_formatted = $view_week_start->format( Dates::DBDATEFORMAT );
+		} else {
+			$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBDATEFORMAT );
+			$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBDATEFORMAT );
+		}
+
+		// If view date is not empty and today does not equal the view date, then do not show.
+		if ( ! empty( $view_date ) && $today_formatted !== $view_date_formatted ) {
+			return;
+		}
+
 		// Flatten Month and Week View Events to Get Accurate Count
-/*		$views_to_flatten = [
+		$views_to_flatten = [
 			'month',
 			'week',
 		];
 
 		if ( in_array( $this->get_slug(), $views_to_flatten ) ) {
 			$events = array_unique( array_merge( ...array_values( $events ) ) );
-		}*/
+		}
 
 		/**
-		 * Filters The Threshold to Show The Recent Past Events.
+		 * Filters The Threshold to Show The Latest Past Events.
 		 * Defaults to show when there are Zero Events.
 		 *
 		 * @since TBD
 		 *
-		 * @param int   The threshold to show The Recent Past Events.
+		 * @param int   The threshold to show The Latest Past Events.
 		 * @param array $events        Array that will be counted to verify if we have events.
 		 * @param array $template_vars An associative array of variables that will be set, and exported, in the template.
 		 * @param View  $instance      The current View object.
 		 */
-		$recent_past_threshold = apply_filters( "tribe_events_views_v2_threshold_to_show_recent_past_events", 0, $events, $template_vars, $this );
-
-		//todo change to get a saved value instead of running the query
-		$upcoming_events_count = tribe_events()->where( 'ends_after', 'now' )->count();
+		$latest_past_threshold = apply_filters( "tribe_events_views_v2_threshold_to_show_latest_past_events", 0, $events, $template_vars, $this );
```

Comment:
:no_entry_sign: String "tribe_events_views_v2_threshold_to_show_latest_past_events" does not require double quotes; use single quotes instead

<sub>hash: 5c069e089ed4c7a3088dfe95048efead</sub>